src/cargo/util/lints.rs

```diff
@@ -136,6 +138,121 @@ impl From<TomlLintLevel> for LintLevel {
     }
 }
 
+const LINTS: &[Lint] = &[IMPLICIT_FEATURES, UNKNOWN_LINTS];
+
+const UNKNOWN_LINTS: Lint = Lint {
+    name: "unknown_lints",
+    desc: "detects unrecognized lint names",
+    groups: &[],
+    default_level: LintLevel::Warn,
+    edition_lint_opts: None,
+};
+
+pub fn unknown_lints(
+    maybe_pkg: &MaybePackage,
+    path: &Path,
+    resolved_lints: &TomlToolLints,
+    error_count: &mut usize,
+    gctx: &GlobalContext,
+) -> CargoResult<()> {
+    let edition = match maybe_pkg {
+        MaybePackage::Package(pkg) => pkg.manifest().edition(),
+        MaybePackage::Virtual(_) => Edition::default(),
+    };
+
+    let lint_level = UNKNOWN_LINTS.level(resolved_lints, edition);
+    if lint_level == LintLevel::Allow {
+        return Ok(());
+    }
+
+    let original_toml = match maybe_pkg {
+        MaybePackage::Package(pkg) => pkg.manifest().original_toml(),
+        MaybePackage::Virtual(vm) => vm.original_toml(),
+    };
+
+    let contents = match maybe_pkg {
+        MaybePackage::Package(pkg) => pkg.manifest().contents(),
+        MaybePackage::Virtual(vm) => vm.contents(),
+    };
+
+    let document = match maybe_pkg {
+        MaybePackage::Package(pkg) => pkg.manifest().document(),
+        MaybePackage::Virtual(vm) => vm.document(),
+    };
+
+    let renderer = Renderer::styled().term_width(
+        gctx.shell()
+            .err_width()
+            .diagnostic_terminal_width()
+            .unwrap_or(annotate_snippets::renderer::DEFAULT_TERM_WIDTH),
+    );
+    let level = lint_level.to_diagnostic_level();
+    let manifest_path = rel_cwd_manifest_path(path, gctx);
+
+    let all_names = LINTS
+        .iter()
+        .map(|lint| lint.name)
+        .chain(LINT_GROUPS.iter().map(|group| group.name))
+        .collect::<HashSet<_>>();
+
+    if let Some(lints) = original_toml
+        .workspace
+        .as_ref()
+        .map_or(&None, |ws| &ws.lints)
```

Comment:
imo this should be treated as a separate "workspace" lint that is only subject to `workspace.lints` so we have consistency between virtual and real workspace packages.